fe/fe-core/src/main/java/com/starrocks/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -1517,6 +1533,159 @@ public Void visitDefaultValueExpr(DefaultValueExpr node, Scope context) {
+            if (!(params.get(0) instanceof StringLiteral)) {
+                throw new SemanticException("dict_mapping function first param table_name should be string literal");
+            }
+            String[] dictTableFullName = ((StringLiteral) params.get(0)).getStringValue().split("\\.");
+            TableName tableName;
+            if (dictTableFullName.length == 1) {
+                tableName = new TableName(null, dictTableFullName[0]);
+                tableName.normalization(session);
+            } else if (dictTableFullName.length == 2) {
+                tableName = new TableName(dictTableFullName[0], dictTableFullName[1]);
+            } else {
+                throw new SemanticException("dict_mapping function first param table_name should be 'db.tbl' or 'tbl' format");
+            }
+
+            Database db = GlobalStateMgr.getCurrentState().getDb(tableName.getDb());
+            if (db == null) {
+                throw new SemanticException("Database %s is not found", tableName.getDb());
+            }
+            Table table = db.getTable(tableName.getTbl());
+            if (table == null) {
+                throw new SemanticException("dict table %s is not found", tableName.getTbl());
+            }
+            if (!(table instanceof OlapTable)) {
+                throw new SemanticException("dict table type is not OlapTable, type=" + table.getClass());
+            }
+            if (table instanceof MaterializedView) {
+                throw new SemanticException("dict table can't be materialized view");
+            }
+            OlapTable dictTable = (OlapTable) table;
+
+            if (dictTable.getKeysType() != KeysType.PRIMARY_KEYS) {
+                throw new SemanticException("dict table " + tableName + " should be primary key table");
+            }
+
+            // verify keys length and type
+            List<Column> keyColumns = new ArrayList<>();
+            Column valueColumn = null;
+            for (Column column : dictTable.getBaseSchema()) {
+                if (column.isKey()) {
+                    keyColumns.add(column);
+                }
+                if (column.isAutoIncrement()) {
+                    valueColumn = column;
+                }
+            }
+            // (table, keys..., value_column, strict_mode)
+            int valueColumnIdx;
+            int strictModeIdx;
+            if (params.size() == keyColumns.size() + 1) {
+                valueColumnIdx = -1;
+                strictModeIdx = -1;
+            } else if (params.size() == keyColumns.size() + 2) {
+                if (params.get(params.size() - 1).getType().getPrimitiveType().isStringType()) {
+                    valueColumnIdx = params.size() - 1;
+                    strictModeIdx = -1;
+                } else {
+                    strictModeIdx = params.size() - 1;
+                    valueColumnIdx = -1;
+                }
+            } else if (params.size() == keyColumns.size() + 3) {
```

Comment:
When will the `params.size() == keyColumns.size() + 3`?

Reply:
DICT_MAPPING(table_name, keys * N, value_column_name, strict_mode)
All args are keys size and other 3 params.